### ISSUE TITLE
The software logical

### ISSUE THREAD

**Issue by dabaicai521** [joyjitchatterjee/WindTurbine-QAKG#1]
Can you provide detailed software logic or instructions?

**Comment by joyjitchatterjee** [comment]
Hi @dabaicai521 ,

Thank you for your comment. The usage is directly based on the Neo4j KG Database Management System (https://neo4j.com/docs/getting-started/current/). The key essence is: 

- In the [WindTurbine-QAKG](https://github.com/joyjitchatterjee/WindTurbine-QAKG)/data/ folder, the script in [CypherScript_XAI4Wind.txt](https://github.com/joyjitchatterjee/WindTurbine-QAKG/blob/main/data/CypherScript_XAI4Wind.txt) shows the Cypher queries which can be used for creating a domain-specific knowledge graph for operations and maintenance of wind turbines. This code can directly be used in Neo4j for creating and replicating the Knowledge Graph we have already made, or for modifying it and updating it with new information.
- [Final_Paraphrased_WindTurbine-Latest.csv](https://github.com/joyjitchatterjee/WindTurbine-QAKG/blob/main/data/Final_Paraphrased_WindTurbine-Latest.csv) provides a file with Questions-Cypher Query pairs. You can use Questions as Inputs and Cypher Query Pairs as Targets for various purposes and applications. For example, an AI model can be developed for predicting Cypher queries based on natural language questions provided by the users. This would be similar to a neural machine translation (more specifically, a formal language generation) system.
- The remaining files pertain to how the development was done for the KG based on explicit templates and queries created from domain-specific maintenance manuals in the wind industry e.g. the Python script used for creating standardised templates of Questions and Cypher Query pairs. 

More details on the software are available at: https://neo4j.com/docs/getting-started/current/, which can be downloaded to use the Cypher query script. 

In summary - [CypherScript_XAI4Wind.txt](https://github.com/joyjitchatterjee/WindTurbine-QAKG/blob/main/data/CypherScript_XAI4Wind.txt)  can be used for creating or loading the KG described in this repository. While, [Final_Paraphrased_WindTurbine-Latest.csv](https://github.com/joyjitchatterjee/WindTurbine-QAKG/blob/main/data/Final_Paraphrased_WindTurbine-Latest.csv) is more of a dataset (resource) that can be used for formal language generation and interactive question-answering with AI. 

Thanks!
Joyjit

**Comment by dabaicai521** [comment]
Thanks!

**Comment by joyjitchatterjee** [comment]
Closing this issue as instructions have now been updated in the repo as requested.
